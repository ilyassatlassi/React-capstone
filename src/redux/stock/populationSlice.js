import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

export const fetchPopulationData = createAsyncThunk(
  'population/fetchPopulationData',
  async () => {
    const response = await axios.get('https://restcountries.com/v3.1/all');
    return response.data.map((country) => ({
      name: country.name.common,
      population: country.population,
      flag: country.flags.png,
      capital: country.capital,
      region: country.region,
      subregion: country.subregion,
      // currencies:country.currencies.name,
      // symbol:country.currencies.symbol,

    }));
  },
);

export const populationSlice = createSlice({
  name: 'population',
  initialState: {
    data: [],
    status: 'idle',
    error: null,
  },
  reducers: {
    setData: (state, action) => {
      /* eslint-disable */
      state.data = action.payload;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchPopulationData.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchPopulationData.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.data = action.payload;
      })
      .addCase(fetchPopulationData.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.error.message;
      });
  },
});
/* eslint-disable */

export const selectFilteredData = (state) => state.population.data;

export const { setData } = populationSlice.actions;

export default populationSlice.reducer;
