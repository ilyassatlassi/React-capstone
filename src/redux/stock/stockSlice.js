import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import axios from "axios";

const url =
  "https://financialmodelingprep.com/api/v3/balance-sheet-statement/AAPL?apikey=6f672427883fe56dd0c35cb0504f7bda&limit=120";

export const getStock = createAsyncThunk("stock/getStock", async () => {
  try {
    const resp = await axios.get(url);
    const data = resp.data;

    return data;
  } catch (error) {
    return error.message;
  }
});

const initialState = {
  stock: [],
  isLoading: false,
  error: undefined,
};

const stockSlice = createSlice({
    name: 'books',
    initialState,
    reducers: { },
    extraReducers: (builder) => {
      builder
        .addCase(getStock.pending, (state) => ({
          ...state,
          isLoading: true,
        }))
        .addCase(getStock.fulfilled, (state, action) => {
            console.log(action.payload); // Add this console.log statement
            return {
              ...state,
              stock: action.payload,
              isLoading: false,
            }
          })
          
        .addCase(getStock.rejected, (state, action) => ({
          ...state,
          isLoading: false,
          error: action.payload,
        }))
    },
  
  });

  export default stockSlice.reducer;