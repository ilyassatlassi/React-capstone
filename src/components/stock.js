import { getStock } from "../redux/stock/stockSlice"
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from "react";


export default function Stock(){
    const { stock } = useSelector((state) => state.stock);
    const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getStock());
  }, [dispatch]);
  console.log(stock)


}