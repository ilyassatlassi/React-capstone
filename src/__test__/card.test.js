import React from 'react';
import { render } from '@testing-library/react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Info from '../components/moreInfo';
import { selectFilteredData} from '../redux/stock/populationSlice';

jest.mock('react-router-dom', () => ({
  useParams: jest.fn()
}));

jest.mock('react-redux', () => ({
  useSelector: jest.fn()
}));

describe('Info', () => {
  beforeEach(() => {
    useParams.mockReturnValue({ countryName: 'United States' });
    useSelector.mockReturnValue([
      {
        name: 'United States',
        flag: 'https://restcountries.com/data/usa.svg',
        capital: 'Washington, D.C.',
        region: 'Americas',
        subregion: 'Northern America'
      }
    ]);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when character data is available', () => {
    const { container } = render(<Info />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly when character data is not available', () => {
    useSelector.mockReturnValue([]);

    const { container } = render(<Info />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
