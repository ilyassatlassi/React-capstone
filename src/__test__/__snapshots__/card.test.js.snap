// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders correctly when character data is available 1`] = `
<div
  class="info"
>
  <div
    class="selected-country"
  >
    <img
      alt="United States flag"
      class="country-flag-info"
      src="https://restcountries.com/data/usa.svg"
    />
    <h2>
      Name:
      United States
    </h2>
    <p>
      Capital:
      Washington, D.C.
    </p>
    <p>
      Region:
      Americas
    </p>
    <p>
      Subregion:
      Northern America
    </p>
  </div>
</div>
`;

exports[`Info renders correctly when character data is not available 1`] = `
<div
  class="info"
>
  <div>
    No data available for 
    United States
  </div>
</div>
`;
